### PR TITLE
feat(*): allow setting min and max approvals

### PR DESCRIPTION
**Pull request by FiboApe**

### CODE REVIEW

File: apps/root/src/frame/components/navigation/index.tsx [hunk 1]
```diff
@@ -357,6 +364,16 @@ const Navigation = ({ children }: React.PropsWithChildren) => {
             type: OptionsMenuOptionType.option,
           },
           ...languageOptions,
+          {
+            label: intl.formatMessage(
+              defineMessage({ description: 'useUnlimitedApproval', defaultMessage: 'Unlimited Approval' })
```
Comment (th0rOdinson) [inline apps/root/src/frame/components/navigation/index.tsx:369]: Maybe a tooltip with some info would be useful here?

File: apps/root/src/common/components/transaction-steps/index.tsx [hunk 2]
```diff
@@ -485,27 +479,25 @@ const buildApproveTokenItem = ({
...
-                  variant="outlined"
-                  disabled={isPendingTransaction}
-                >
-                  {specificBtnTextAsSecondary}
-                </Button>
+            <>
+              {isCurrentStep && (
+                <ContainerBox gap={3}>
+                  <Button
+                    onClick={isPermit2Enabled && useUnlimitedApproval ? () => onAction() : () => onAction(amount)}
```
Comment (th0rOdinson) [inline apps/root/src/common/components/transaction-steps/index.tsx:486]: isPermit2Enabled i think comes from agg settings, shouldn't we erase that setting?

Comment (FiboApe) [reply apps/root/src/common/components/transaction-steps/index.tsx:486]: Screenshot 2024-11-07 at 5 50 29 pm
We actually use it quite a bit

File: apps/root/src/frame/components/navigation/index.tsx (hunk 1 above)
Comment (FiboApe) [reply apps/root/src/frame/components/navigation/index.tsx:369]: We dont actually allow tooltips in this items, in any case, its a feature for more expert users that already know what it means, and for more common users I dont want them even touching it :P

Comment (FiboApe) [reply apps/root/src/frame/components/navigation/index.tsx:369]: Btw tried with a secondary label and it just looks too big

**Review by th0rOdinson — APPROVED** [review]

**Pull request merged**